src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModelKinDynWrapper.h

```diff
@@ -0,0 +1,165 @@
+     * @param tauExt a vector of size number of joints containing the joint torques generated by the
+     * external contacts.
+     * @param baseAcceleration the sub-model base acceleration.
+     * @return a boolean value
+     */
+    bool inverseDynamics(Eigen::VectorXd& motorTorqueAfterGearbox,
+                         Eigen::VectorXd& frictionTorques,
+                         Eigen::VectorXd& tauExt,
+                         Eigen::VectorXd& baseAcceleration,
+                         Eigen::VectorXd& jointAcceleration);
+
+    /**
+     * @brief getBaseAcceleration gets the acceleration of the sub-model base
+     * @param kinDynFullModel is the KinDynComputation object associated to a model.
+     * @param robotBaseAcceleration the acceleration of the robot base.
+     * @param robotJointAcceleration the acceleration of the robot joints.
+     * @param subModelBaseAcceleration the base acceleration of the sub-model.
+     * @return a boolean value.
+     */
+    bool getBaseAcceleration(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel,
+                             const Eigen::VectorXd& robotBaseAcceleration,
+                             const Eigen::VectorXd& robotJointAcceleration,
+                             Eigen::VectorXd& subModelBaseAcceleration);
+
+    /**
+     * @brief getBaseVelocity gets the acceleration of the sub-model base
+     * @param kinDynFullModel is the KinDynComputation object associated to a model.
+     * @return the the base velocity of the sub-model.
+     */
+    Eigen::VectorXd getBaseVelocity(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
+
+    /**
+     * Getters
+     */
+
+    /** Access massMatrix
+     * @return The current value of massMatrix
+     */
+    iDynTree::MatrixDynSize getMassMatrix();
+
+    /** Access genForces
+     * @return The current value of genForces
+     */
+    iDynTree::FreeFloatingGeneralizedTorques getGeneralizedForces();
+
+    /** Access jacFT
+     * @return The current value of jacFT
+     */
+    iDynTree::MatrixDynSize getFTJacobian(const std::string& ftName);
+
+    /** Access jacAcc
+     * @return The current value of jacAcc
+     */
+    iDynTree::MatrixDynSize getAccelerometerJacobian(const std::string& accName);
+
+    /** Access jacGyro
+     * @return The current value of jacGyro
+     */
+    iDynTree::MatrixDynSize getGyroscopeJacobian(const std::string& gyroName);
+
+    /** Access jacContact
+     * @return The current value of jacContact
+     */
+    iDynTree::MatrixDynSize getExtContactJacobian(const std::string& extContactName);
+
+    /** Access dJnu
+     * @return The current value of dJnu
+     */
+    iDynTree::Vector6 getAccelerometerBiasAcceleration(const std::string& accName);
+
+    /** Access accRworld
+     * @return The current value of accRworld
+     */
+    iDynTree::Matrix3x3 getAccelerometerRotation(const std::string& accName);
```

Comment:
All these outputs can be turned into const references to avoid memory allocation

Reply:
I just noticed I did not implement the getters :face_in_clouds: I'm going to add their implementation!